src/Microsoft.OpenApi.Readers/OpenApiStreamReader.cs

```diff
@@ -29,13 +29,20 @@ public OpenApiStreamReader(OpenApiReaderSettings settings = null)
         /// Reads the stream input and parses it into an Open API document.
         /// </summary>
         /// <param name="input">Stream containing OpenAPI description to parse.</param>
-        /// <param name="diagnostic">Returns diagnostic object containing errors detected during parsing</param>
-        /// <returns>Instance of newly created OpenApiDocument</returns>
+        /// <param name="diagnostic">Returns diagnostic object containing errors detected during parsing.</param>
+        /// <returns>Instance of newly created OpenApiDocument.</returns>
         public OpenApiDocument Read(Stream input, out OpenApiDiagnostic diagnostic)
         {
-            using (var reader = new StreamReader(input))
+            if (_settings.LeaveStreamOpen)
```

Comment:
```CSharp
var reader = new StreamReader(input);
var result = new OpenApiTextReaderReader(_settings).Read(reader, out diagnostic);
if (!_settings.LeaveStreamOpen)
   reader.Dispose();
return result;
```

Reply:
Yes. Much cleaner!

Reply:
I like brackets after `if` statements though, even for one-liners 😄

Reply:
I won't fight coding preferences. It seems that I'm the only one on the team to like one liners with no brackets 😅